wit/types.wit

```diff
@@ -0,0 +1,29 @@
+@unstable(feature = tls)
+interface types {
+    @unstable(feature = tls)
+    use wasi:io/streams@0.2.3.{input-stream, output-stream};
+    @unstable(feature = tls)
+    use wasi:io/poll@0.2.3.{pollable};
+
+    @unstable(feature = tls)
+    resource client-handshake {
+        @unstable(feature = tls)
+        constructor(server-name: string, input: input-stream, output: output-stream);
+
+        @unstable(feature = tls)
+        finish: static func(this: client-handshake) -> future-client-streams;
+    }
+
+    @unstable(feature = tls)
+    resource client-connection {
+    }
+
+    @unstable(feature = tls)
+    resource future-client-streams {
+        @unstable(feature = tls)
+        subscribe: func() -> pollable;
+
+        @unstable(feature = tls)
+        get: func() -> option<result<result<tuple<client-connection, input-stream, output-stream>>>>;
```

Comment:
I imagine we'll want to define a `variant error` type with cases covering at least the most common failure cases (e.g. expired certificate, hostname mismatch, etc.) and use it here, eventually.

Reply:
Yup, definitely. 👍 

I wasn't in the mood to do a full inventory of all the ways a TLS connection may fail just yet, so I lazily wrote it off as a problem for future me :sweat_smile: